### ISSUE TITLE
Datatable in cell edit with calendar navigator bug

### ISSUE THREAD

**Issue by jxmai** [primefaces/primefaces#1265]
The same issue has been posted in the primefaces forum that marked as cannot reproduce:
http://forum.primefaces.org/viewtopic.php?f=3&t=28075

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058139/41156674-f2e5-11e5-815e-9bffba0e96cc.png)

**A click in the date-picker causes a blur of the cell editor. The cell editor switchs from the input mode to the output mode too soon.**

Here is the simple complete testing example:

**edit.xhtml**

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:ui="http://java.sun.com/jsf/facelets"
    xmlns:h="http://java.sun.com/jsf/html"
    xmlns:f="http://java.sun.com/jsf/core"
    xmlns:p="http://primefaces.org/ui"
    xmlns:fn="http://java.sun.com/jsp/jstl/functions">

<h:head>

</h:head>

<h:body>
    <h:form id="form">
        <p:growl id="msgs" showDetail="true" />

        <p:dataTable value="#{editableTableBean.persons}"
                     var="person"
                     editable="true"
                     editMode="cell"
        >
            <f:facet name="header">
                Celledit with calendar Bug
            </f:facet>

            <p:column headerText="Name">

                <p:cellEditor>
                    <f:facet name="output">
                        <p:outputLabel value="#{person.name}"></p:outputLabel>
                    </f:facet>

                    <f:facet name="input">
                        <p:inputText value="#{person.name}"></p:inputText>
                    </f:facet>
                </p:cellEditor>
            </p:column>

            <p:column headerText="Date of Birth">
                <p:cellEditor>
                    <f:facet name="output">
                        <p:outputLabel value="#{person.dateOfBirth}"></p:outputLabel>
                    </f:facet>
                    <f:facet name="input">
                        <p:calendar value="#{person.dateOfBirth}"
                                    navigator="true"
                        >
                        </p:calendar>
                    </f:facet>
                </p:cellEditor>
            </p:column>
        </p:dataTable>
    </h:form>
</h:body>

</html>
```

**EditableTableBean.java**

```
import java.io.Serializable;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.List;

import javax.annotation.PostConstruct;
import javax.faces.bean.ManagedBean;
import javax.faces.bean.ViewScoped;

@ManagedBean(name="editableTableBean")
@ViewScoped
public class EditableTableBean implements Serializable {

    private static final long serialVersionUID = -4217194079931366514L;
    private List<Person> persons = new ArrayList<Person>();

    @PostConstruct
    public void init() {
        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");

        Person person1;
        Person person2;
        try {
            person1 = new Person("John", sdf.parse("01/01/1979"));
            person2 = new Person("Rose", sdf.parse("01/01/1969"));

            this.getPersons().add(person1);
            this.getPersons().add(person2);
        } catch (ParseException e) {
            e.printStackTrace();
        }
    }

    public List<Person> getPersons() {
        return persons;
    }

    public void setPersons(List<Person> persons) {
        this.persons = persons;
    }

}

```

**Person.java**

```
import java.io.Serializable;
import java.util.Date;

public class Person implements Serializable {

    /**
     * 
     */
    private static final long serialVersionUID = 7247747818315769598L;
    private String name;
    private Date dateOfBirth;

    public Person(String name, Date dateOfBirth) {
        super();
        this.name = name;
        this.dateOfBirth = dateOfBirth;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getDateOfBirth() {
        return dateOfBirth;
    }

    public void setDateOfBirth(Date dateOfBirth) {
        this.dateOfBirth = dateOfBirth;
    }

}

```

**_Running the example**_
1. The inital screen shot:

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058167/3a44413e-f2e6-11e5-987f-9157b4f2d8e2.png)
1. Change the birthday of John within the same month works ok

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058189/9bc46a60-f2e6-11e5-88e7-fbc53e3eebd4.png)

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058198/d4a2a3b0-f2e6-11e5-845c-3c5bfdaa4c65.png)

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058200/f1c58f66-f2e6-11e5-9def-717e6823e708.png)
1. Change the birthday of John to a  different month does not work because the output mode fire too soon.

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058213/586fa936-f2e7-11e5-8e1b-f68a3faeb1e3.png)

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058215/75c3d890-f2e7-11e5-91d7-6d17e3fc420a.png)

The date did not change.

A possible solution for this issue is to add one more condition(**_!$.datepicker._datepickerShowing**_) in the datatable.js file:

![screenshot](https://cloud.githubusercontent.com/assets/10467831/14058231/f9ca9d40-f2e7-11e5-85ed-54ad154721c3.png)

This issue solved after I made the change above.

It is mentioned in the last post of the forum I referenced on the top:

```
  $(document).off('click.datatable-cell-blur' + this.id)
                        .on('click.datatable-cell-blur' + this.id, function(e) {
                            if(!$this.incellClick && $this.currentCell && !$this.contextMenuClick && !$.datepicker._datepickerShowing) {
                                $this.saveCell($this.currentCell);
                            } ...
```

**Comment by mertsincan** [comment]
Thanks a lot for your PR ;) +1 
Fixed for 5.3.10 and 6.0.

**Comment by jxmai** [comment]
No problem. I am glad it helps. :+1:
